gravitee-am-repository/gravitee-am-repository-jdbc/src/main/java/io/gravitee/am/repository/jdbc/management/api/JdbcRateLimitRepository.java

```diff
@@ -0,0 +1,150 @@
+
+    protected RateLimit toEntity(JdbcRateLimit entity) { return mapper.map(entity, RateLimit.class);}
+    protected JdbcRateLimit toJdbcEntity(RateLimit entity) {return mapper.map(entity, JdbcRateLimit.class);}
+
+    @Override
+    public Maybe<RateLimit> findById(String id) {
+        LOGGER.debug("RateLimit findById({})", id);
+        return rateLimitRepository.findById(id)
+                .map(this::toEntity);
+    }
+
+    @Override
+    public Maybe<RateLimit> findByCriteria(RateLimitCriteria criteria) {
+        Criteria whereClause = buildWhereClause(criteria);
+        return monoToMaybe(template.select(Query.query(whereClause).with(PageRequest.of(0,1, Sort.by("id"))), JdbcRateLimit.class)
+                .singleOrEmpty()).map(this::toEntity);
+    }
+
+    @Override
+    public Single<RateLimit> create(RateLimit item) {
+        item.setId(item.getId() == null ? RandomString.generate() : item.getId());
+        LOGGER.debug("create RateLimit with id {}", item.getId());
+        return monoToSingle(template.insert(toJdbcEntity(item))).map(this::toEntity);
+    }
+
+    @Override
+    public Single<RateLimit> update(RateLimit item) {
+        LOGGER.debug("update RateLimit with id '{}'", item.getId());
+        return rateLimitRepository.save(toJdbcEntity(item))
+                .map(this::toEntity);
+    }
+
+    @Override
+    public Completable delete(String id) {
+        LOGGER.debug("delete RateLimit with id '{}'", id);
+        return rateLimitRepository.deleteById(id);
+    }
+
+    @Override
+    public Completable delete(RateLimitCriteria criteria) {
+        LOGGER.debug("delete RateLimit with id '{}'", criteria);
+        Criteria whereClause = buildWhereClause(criteria);
+
+        if (!whereClause.isEmpty()) {
+            return monoToCompletable(template.delete(JdbcRateLimit.class).matching(Query.query(whereClause)).all());
+        }
+
+        throw new RepositoryIllegalQueryException("Unable to delete from RateLimit with criteria");
+    }
+
+    @Override
+    public Completable deleteByUser(String userId) {
+        LOGGER.debug("delete RateLimit with user id '{}'", userId);
+
+        Criteria whereClause = Criteria.empty();
+        if(userId != null && !userId.isEmpty()){
+            whereClause = whereClause.and(where("user_id").is(userId));
+        }
+
+        if (!whereClause.isEmpty()) {
+            return monoToCompletable(template.delete(JdbcRateLimit.class).matching(Query.query(whereClause)).all());
+        }
+
+        throw new RepositoryIllegalQueryException("Unable to delete from RateLimit with userId");
+    }
+
+    @Override
+    public Completable deleteByDomain(String domainId) {
+        LOGGER.debug("delete RateLimit with domain id '{}'", domainId);
+
+        Criteria whereClause = Criteria.empty();
+        if(domainId != null && !domainId.isEmpty()){
+            whereClause = whereClause.and(where("reference_id").is(domainId));
```

Comment:
Maybe we can add ReferenceType.DOMAIN as a second criteria

Reply:
done, added as suggested